yroots/subdivision.py

```diff
@@ -620,76 +632,76 @@ def subdivision_solve_nd(funcs,a,b,deg,target_deg,interval_data,root_tracker,tol
         good_degs = [None]*len(funcs)
     inf_norms = []
     approx_errors = []
-    #Get the chebyshev approximations
+    # Get the chebyshev approximations
     for func, good_deg in zip(funcs, good_degs):
-        coeff,change_sign,inf_norm,approx_error = full_cheb_approximate(func,a,b,deg,tols.abs_approx_tol,tols.rel_approx_tol, good_deg)
+        if use_target_tol:
+            coeff,change_sign,inf_norm,approx_error = full_cheb_approximate(func,a,b,deg,tols.target_tol,tols.rel_approx_tol, good_deg)
+        else:
+            coeff,change_sign,inf_norm,approx_error = full_cheb_approximate(func,a,b,deg,tols.abs_approx_tol,tols.rel_approx_tol, good_deg)
         inf_norms.append(inf_norm)
         approx_errors.append(approx_error)
-        #Subdivides if a bad approximation
+        # Subdivides if a bad approximation
         if coeff is None:
             intervals = get_subintervals(a,b,change_sign,None,None,None,approx_errors)
             for new_a, new_b in intervals:
                 subdivision_solve_nd(funcs,new_a,new_b,deg,target_deg,interval_data,root_tracker,tols,max_level,level=level+1, method=method)
             return
         else:
-            #if the function changes sign on at least one subinterval, skip the checks
+            # if the function changes sign on at least one subinterval, skip the checks
             if np.any(change_sign):
                 cheb_approx_list.append(coeff)
                 continue
-            #Run checks to try and throw out the interval
+            # Run checks to try and throw out the interval
             if interval_data.check_interval(coeff, approx_error, a, b):
                 return
 
             cheb_approx_list.append(coeff)
 
-    #Make the system stable to solve
+    # Make the system stable to solve
     coeffs, good_approx, approx_errors = trim_coeffs(cheb_approx_list, tols.abs_approx_tol, tols.rel_approx_tol, inf_norms, approx_errors)
 
-    #Used if subdividing further.
+    # Used if subdividing further.
     good_degs = [coeff.shape[0] - 1 for coeff in coeffs]
     good_zeros_tol = max(tols.min_good_zeros_tol, np.sum(np.abs(approx_errors))*tols.good_zeros_factor)
+        
+    # Runs the same things as above, but we want to get rid of that eventually so keep them seperate.
```

Comment:
I don't understand this. What things are we running again and why do we want to get rid of that?

Reply:
I believe this was leftover from a previous version of the code, so I don't think it makes any sense in this context. I'll update the comment to be more helpful.